homeassistant/components/vodafone_station/coordinator.py

```diff
@@ -94,16 +94,20 @@ def _calculate_update_time_and_consider_home(
     async def _async_update_data(self) -> UpdateCoordinatorDataType:
         """Update router data."""
         _LOGGER.debug("Polling Vodafone Station host: %s", self._host)
+        logged = False
         try:
             logged = await self.api.login()
         except exceptions.CannotConnect as err:
             _LOGGER.warning("Connection error for %s", self._host)
+            await self.api.close()
             raise UpdateFailed(f"Error fetching data: {repr(err)}") from err
-        except exceptions.CannotAuthenticate as err:
-            raise ConfigEntryAuthFailed from err
-
-        if not logged:
-            raise ConfigEntryAuthFailed
+        except exceptions.AlreadyLogged as err:
+            _LOGGER.warning("User %s already logged. Retrying", self.api.username)
+            await self.api.close()
+            raise UpdateFailed(f"Error fetching data: {repr(err)}") from err
```

Comment:
This will end up raising both UpdateFailed and ConfigEntryAuthFailed. The finally clause will always execute.

Reply:
Fixed